akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -178,6 +178,16 @@ object Source {
   private[this] val _empty: Source[Nothing] = apply(EmptyPublisher)
 
   /**
+   * A `Source` with no elements and does not automatically `onComplete`.
+   *
+   * It materializes a [[scala.concurrent.Promise]] which will be completed
+   * when the downstream stage of this source cancels. This promise can also
+   * be used to externally trigger completion, which the source then signalls
+   * to its downstream.
+   */
+  def never[T]() = NeverSource[T]()
```

Comment:
I am not sure about the name. KeepAliveEmtpySource is long, maybe someone has a better idea?